### PR TITLE
Add `cacheTo` argument to `ci` action

### PR DESCRIPTION
**Pull request by sebst**
Added the argument `cacheTo` to the action (#299) so that a cache image can be written to a registry.

### CODE REVIEW

**Comment by sebst** [comment]
@microsoft-github-policy-service agree

File: azdo-task/DevcontainersCi/task.json [hunk 1]
```diff
@@ -122,6 +122,12 @@
       "type": "boolean",
       "label": "Builds the image with `--no-cache` (takes precedence over `cacheFrom`)",
       "required": false
+    },
+    {
+      "name": "cacheTo",
```
Comment (chrmarti) [inline azdo-task/DevcontainersCi/task.json:127]: Also document in docs/azure-devops-task.md.

File: action.yml [hunk 2]
```diff
@@ -63,6 +63,9 @@ inputs:
     required: false
     default: false
     description: Builds the image with `--no-cache` (takes precedence over `cacheFrom`)
+  cacheTo:
```
Comment (chrmarti) [inline action.yml:66]: Also document in docs/github-action.md

File: github-action/src/main.ts [hunk 3]
```diff
@@ -57,6 +57,7 @@ export async function runMain(): Promise<void> {
 		const inputEnvsWithDefaults = populateDefaults(inputEnvs, inheritEnv);
 		const cacheFrom: string[] = core.getMultilineInput('cacheFrom');
 		const noCache: boolean = core.getBooleanInput('noCache');
+		const cacheTo = emptyStringAsUndefined(core.getInput('cacheTo'));
```
Comment (chrmarti) [inline github-action/src/main.ts:60]: Support array?

**Review by chrmarti — CHANGES_REQUESTED** [review]
Thanks for the PR! Left a few comments.

**Comment by sebst** [comment]
Any update?

**Comment by sebst** [comment]
Any update?

File: github-action/src/main.ts [hunk 4]
```diff
@@ -57,6 +57,7 @@ export async function runMain(): Promise<void> {
 		const inputEnvsWithDefaults = populateDefaults(inputEnvs, inheritEnv);
 		const cacheFrom: string[] = core.getMultilineInput('cacheFrom');
 		const noCache: boolean = core.getBooleanInput('noCache');
+		const cacheTo: string[] = core.getMultilineInput('cacheFrom');
```
Comment (samruddhikhandale) [inline github-action/src/main.ts:60]:
```suggestion
		const cacheTo: string[] = core.getMultilineInput('cacheTo');
```

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Thanks, overall it looks good.

Left one comment

**Comment by sebst** [comment]
> Thanks, overall it looks good.
> 
> Left one comment

Thanks! Fixed the typo

**Review by samruddhikhandale — APPROVED** [review]

**Comment by samruddhikhandale** [comment]
> Ensure you have run scripts/build-local.sh

@sebst Can you help fix the failing action, see [here](https://github.com/devcontainers/ci/actions/runs/11244625749/job/31263435255?pr=300#step:10:2223)

**Comment by sebst** [comment]
> > Ensure you have run scripts/build-local.sh
> 
> @sebst Can you help fix the failing action, see [here](https://github.com/devcontainers/ci/actions/runs/11244625749/job/31263435255?pr=300#step:10:2223)

Done

**Comment by sebst** [comment]
@samruddhikhandale could you please restart the pipeline to see if the last commit fixes the issue?

**Comment by sebst** [comment]
Can this be merged now?

@samruddhikhandale @chrmarti @stuartleeks @bamurtaugh

**Review by samruddhikhandale — APPROVED** [review]

**Comment by samruddhikhandale** [comment]
Hi @sebst, thanks for your patience. 
We are waiting for another review from @chrmarti, we'd merge shortly. Thanks!

**Comment by sebst** [comment]
Just a friendly reminder. 

@samruddhikhandale @chrmarti @stuartleeks @bamurtaugh

**Comment by sebst** [comment]
Just another friendly reminder.

@samruddhikhandale @chrmarti @stuartleeks @bamurtaugh

**Comment by sebst** [comment]
> Hi @sebst, thanks for your patience. 
> 
> We are waiting for another review from @chrmarti, we'd merge shortly. Thanks! 

Is there any chance we get that merged?

@chrmarti

**Comment by sebst** [comment]
@samruddhikhandale @chrmarti @stuartleeks @bamurtaugh

This PR is open for more than 3 months now but it's just a small change.

Is there anything I could do on my end to get this merged?

Appreciate any feedback!

**Review by chrmarti — APPROVED** [review]
LGTM, thanks!

**Pull request merged**